### PR TITLE
Add tests for Thread constructor with stack size argument

### PR DESCRIPTION
**Pull request by kouvel**
Fixes dotnet/coreclr#9158
Fixes dotnet/coreclr#9167

Part of dotnet/coreclr#9186

### CODE REVIEW

**Comment by kouvel** [comment]
@janvorli @jkotas

File: src/System.Threading.Thread/tests/ThreadTests.netstandard.cs [hunk 1]
```diff
@@ -20,10 +20,21 @@ public static class ThreadTests
...
+                t =>
+                {
+                    t.IsBackground = true;
+                    t.Start();
+                    t.Join();
+                };
+            startThreadAndJoin(new Thread(() => { }));
+            startThreadAndJoin(new Thread(() => { }, 0));
+            startThreadAndJoin(new Thread(() => { }, 1));
+            startThreadAndJoin(new Thread(() => { }, 16 << 20));
```
Comment (JonHanna) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard.cs]: Maybe make this a `[Theory]` and pass in sizes, with the invalid and default cases separate tests?

Comment (kouvel) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard.cs]: I tried it but it looks far less readable. As I'd like the constructor call to be in the test and not in the member data generation, and I'm testing different constructors here, it doesn't seem worth having 4 separate tests of which 2 would be theories for something trivial like this. If this fails I'm fine with it stopping at the first failure.

**Comment by karelz** [comment]
Oh darn, I hit merge, despite the fact it was NO MERGE, sorry!

**Comment by karelz** [comment]
@kouvel I merged the PR, because the tests passed and there was code review approval.
They should have failed until the dependencies in coreclr were merged -- do you have explanation why the tests passed? (Bad coverage, CI infra issues, or anything like that)

I won't revert the merge unless I see issues in the CI caused by it ... (discussed offline with @stephentoub) ... @kouvel feel free to do what you seem is best here. Sorry for the trouble :(

**Pull request merged**